singer_sdk/exceptions.py

```diff
@@ -11,6 +11,21 @@
 class ConfigValidationError(Exception):
     """Raised when a user's config settings fail validation."""
 
+    def __init__(
+        self,
+        message: str,
+        *,
+        errors: list[str] | None = None,
```

Comment:
splitting hairs here, but can't this be
```suggestion
        errors: list[str] = [],
```
so that types are a tiny bit more constrained?

Reply:
Though that would be nice, assigning a mutable object like a list as a default value for a function parameter means that every call shares that default object. If the function every modifies the list, then those modifications will apply to every future call.

In a future version of Python this will hopefully be fixed by being able to write `errors: list[str] => []`, which will result in the default value being recomputed on every call, instead of when the function is defined. See [PEP 671](https://peps.python.org/pep-0671/).

Reply:
This is a very good point! If forget this point way too often :( Thanks for pointing it out!